src/include/platform/internal/DeviceControlServer.h

```diff
@@ -0,0 +1,69 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *          Defines the Device Layer DeviceControlServer object.
+ */
+
+#pragma once
+
+#include <platform/internal/CHIPDeviceLayerInternal.h>
+
+namespace chip {
+namespace DeviceLayer {
+namespace Internal {
+
+class DeviceControlServer final
+{
+public:
+    // ===== Members for internal use by other Device Layer components.
+
+    CHIP_ERROR HandleArmFailSafe(uint16_t expiryLengthSeconds);
+    CHIP_ERROR HandleDisarmFailSafe(void);
+    CHIP_ERROR HandleCommissioningComplete(void);
+    CHIP_ERROR HandleSetRegulatoryConfig(uint8_t location, uint8_t * countryCode, uint64_t breadcrumb);
```

Comment:
why is the country code a pointer? some documentation of parameters may help.

Reply:
Looking further below: this should probably be "const char *" instead.

I think the type mismatch should be fixed at ember level rather than propagate down. If country code is a null-terminated-string, API should show it as such.

Reply:
Acked.

We also need to fix the codegen to generate the prototype with 'const char * countryCode' instead of 'uint8_t * countryCode' for string type. Will look at this issue in followup PR.

Reply:
Just to make sure, what you would like is:
```
bool emberAfGeneralCommissioningClusterSetRegulatoryConfigCallback(chip::app::Command * commandObj, uint8_t location,
                                                                    uint8_t * countryCode, uint64_t breadcrumb, uint32_t timeoutMs)
```

to be generated as:
```
bool emberAfGeneralCommissioningClusterSetRegulatoryConfigCallback(chip::app::Command * commandObj, uint8_t location,
                                                                    char * countryCode, uint64_t breadcrumb, uint32_t timeoutMs)
```

If so and IIRC the only reason why it is still `uint8_t *` was legacy compatibility with the imported clusters from `Silabs` who were using `uint8_t *`.

Reply:
Yes, this is legacy issue, "const char * countryCode" is even better